tests/providers/amazon/aws/hooks/test_redshift_sql.py

```diff
@@ -137,3 +137,130 @@ def test_get_conn_overrides_correctly(self, mock_connect, conn_params, conn_extr
         ):
             self.db_hook.get_conn()
             mock_connect.assert_called_once_with(**expected_call_args)
+
+
+class TestRedshiftSQLHookConnectionScenario:
+    def test_get_iam_token_without_both_cluster_identifier_and_host(self):
+        """
+        Tests if it raises exception when both cluster_identifier and host are not set in redshift connection.
+        """
+        self.connection = Connection(
+            conn_type="redshift",
+            login=LOGIN_USER,
+            password=LOGIN_PASSWORD,
+            port=LOGIN_PORT,
+            schema=LOGIN_SCHEMA,
+        )
+        self.connection.extra = json.dumps(
+            {
+                "iam": True,
+            }
+        )
+        self.db_hook = RedshiftSQLHook()
+        self.db_hook.get_connection = mock.Mock()
+        self.db_hook.get_connection.return_value = self.connection
+        with pytest.raises(Exception, match="Please set cluster_identifier or host in redshift connection."):
+            self.db_hook.get_uri()
+
+    @mock.patch("airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook.conn")
+    @mock.patch("airflow.providers.amazon.aws.hooks.redshift_sql.redshift_connector.connect")
+    @pytest.mark.parametrize("aws_conn_id", [NOTSET, None, "mock_aws_conn"])
+    def test_get_conn_iam_with_cluster_identifier_without_host(
+        self, mock_connect, mock_aws_hook_conn, aws_conn_id
+    ):
+        self.connection = Connection(
```

Comment:
docstring is missing

Reply:
Changed it [here](https://github.com/apache/airflow/pull/31567/commits/0b106d459f353195f8dd0df6ed5954e54adeea48)